### DISCUSSION TITLE
How to get the matrix in cone-bream projection?

### DISCUSSION THREAD

**Discussion by leishaohu** [astra-toolbox/astra-toolbox#477] [Q&A]
Usually we can get matrix through:
```
astra.projector.projection_geometry(proj_id)
matrix_id = astra.projector.matrix(proj_id)
astra.matrix.get_size(matrix_id)
matrix = astra.matrix.get(matrix_id)
```
But if  the projection is cone-beam, and there is no "proj_id". The code is like this:

```
proj_geom = astra.create_proj_geom('cone', 1.0, 1.0, detector_rows, detector_cols, angles, 
                                   distance_source_origin, distance_origin_detector)
vol_geom = astra.create_vol_geom(volume.shape)
vol_id = astra.data3d.create('-vol', vol_geom, volume)

sinogram_id = astra.data3d.create('-sino', proj_geom)

cfg = astra.astra_dict('FP3D_CUDA')
cfg['VolumeDataId'] = vol_id
cfg['ProjectionDataId'] = sinogram_id

alg_id = astra.algorithm.create(cfg)
astra.algorithm.run(alg_id)

proj_data = astra.data3d.get(sinogram_id)
```

What should I do to get the matrix?
Hoping for your guidence!

**Comment by askorikov** [comment]
Hi! Take a look here: #431. Note that it is very computationally expensive and requires a lot of memory to get projection matrices for 3D geometries.

**Reply by leishaohu** [reply]
OK, I understand. Thanks for your answer. I have another question for you: for cone projection, are the rays described in the form of lines or cylinders? How to set these two different projections? I hope to get your guidance.

**Reply by askorikov** [reply]
Rather in the form of lines, but more precisely, slice-interpolated, or Joseph's projection kernel is used (see section 2.1.3 in [this](http://www2.imm.dtu.dk/pubdb/edoc/imm6952.pdf) thesis). This is the only mode supported for 3D geometries at the moment.

**Reply by leishaohu** [reply]
Thank you so much for your help; your guidance has been incredibly useful.

**Reply by askorikov** [reply]
Sure!
